pkg/solana/txm/txm_internal_test.go

```diff
@@ -510,59 +511,6 @@ func TestTxm(t *testing.T) {
 				mc.On("SendTx", mock.Anything, tx).Panic("SendTx should not be called anymore").Maybe()
 			})
 
-			// tx passes sim, shows processed, moves to nil (timeout should cleanup)
```

Comment:
removed this because it's basically a reorg that is tested below

Reply:
Hmm it might still be worth keeping this test to make sure a transaction that doesn't reach confirmed state in time is still cleaned up even if we try to handle re-orgs.

Reply:
Added again. Refactored so instead of regressing it just stays processed and then is cleaned up

Reply:
I think the test `fail_confirm_processed` right above this is already testing this behavior. I think it's ok if we regress to nil and rebroadcast the tx. If the rebroadcasted tx also stalls at processed, we just want to make sure that's cleaned up as well

Reply:
addressed!